dbms/src/Flash/FlashService.cpp

```diff
@@ -593,6 +595,147 @@ grpc::Status FlashService::tryMarkDelete(grpc::ServerContext * grpc_context, con
     return s3_lock_service->tryMarkDelete(request, response);
 }
 
+grpc::Status FlashService::EstablishDisaggregatedTask(grpc::ServerContext * grpc_context, const disaggregated::EstablishDisaggregatedTaskRequest * request, disaggregated::EstablishDisaggregatedTaskResponse * response)
+{
+    CPUAffinityManager::getInstance().bindSelfGrpcThread();
+    LOG_DEBUG(log, "Handling EstablishDisaggregatedTask request: {}", request->ShortDebugString());
+    if (auto check_result = checkGrpcContext(grpc_context); !check_result.ok())
+        return check_result;
+    // TODO metrics
+    auto [db_context, status] = createDBContext(grpc_context);
+    if (!status.ok())
+        return status;
+    db_context->setMockStorage(mock_storage);
+    db_context->setMockMPPServerInfo(mpp_test_info);
+
+    RUNTIME_CHECK(context->isDisaggregatedStorageMode());
```

Comment:
Off topic: When will we use context and when will we use db_context?

Reply:
`context` - actually it is a pointer to the global context
`db_context` - a temporary context that its lifetime lasts only until the end of this request

Reply:
`db_context` is a session-only context, some value of `db_context->getSettings()` is not the same as the global settings.